db/postgres.go

```diff
@@ -0,0 +1,177 @@
+// Copyright © by Jeff Foley 2017-2023. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+// SPDX-License-Identifier: Apache-2.0
+
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"math"
+	"os"
+	"strings"
+
+	migrate "github.com/rubenv/sql-migrate"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+const suggest_upgrade_message = "Upgrade with `amass db upgrade`"
+
+func (p *Postgres) getAppliedMigrationsAmount() int {
+	if p.db.MigrationsTable != "" {
+		migrate.SetTable(p.db.MigrationsTable)
+	}
+
+	sqlDb := p.getSqlConnection()
+
+	appliedMigrations, err := migrate.GetMigrationRecords(sqlDb, "postgres")
+	if err != nil {
+		fmt.Printf("Error getting applied migrations: %s", err)
+		os.Exit(1)
+	}
+
+	return len(appliedMigrations)
+}
```

Comment:
```suggestion
func (p *Postgres) getAppliedMigrationsAmount() (int, error) {
	if p.db.MigrationsTable != "" {
		migrate.SetTable(p.db.MigrationsTable)
	}

	sqlDb := p.getSqlConnection()

	appliedMigrations, err := migrate.GetMigrationRecords(sqlDb, "postgres")
    if err != nil {
		log.f("could not apply migrations due to error: %s ", err)
		# return the error back to the caller so they can handle it
		return 0, err
	}

	return len(appliedMigrations), nil
}
```

so on this, instead of doing frmt.Printf(...)
```golang
func (p *Postgres) getAppliedMigrationsAmount() (int, error)
    // ..

	appliedMigrations, err := migrate.GetMigrationRecords(sqlDb, "postgres")
	if err != nil {
		log.Println("could not apply migrations due to error: ", err)
		return 0, err
	}
    // return the value if it did not error w/ a nil error
    return len(appliedMigrations), nil
}
```